include/hc_printf.hpp

```diff
@@ -82,30 +93,69 @@ static inline void countArg(unsigned int& count, const T& t, const Rest&... rest
   countArg(count,rest...);
 }
 
+static inline unsigned int string_length(const char* str) [[hc,cpu]]{
+  unsigned int size = 0;
+  while(str[size]!='\0')
+    size++;
+  return size;
+}
+
+static inline void copy_n(char* dest, const char* src, unsigned int len) [[hc,cpu]] {
+  for(int i=0; i < len; i++){
+    dest[i] = src[i];
+  }
+}
+
+static inline PrintfError process_str_batch(PrintfPacket* queue, char* fbuffer, int offset, const char* fstring) [[hc,cpu]] {
+  unsigned int fstr_len = string_length(fstring);
+  unsigned int fb_offset = queue[3].data.ai.fetch_add(fstr_len + 1);
+  if (queue[3].data.ui > queue[2].data.ui){
+    return PRINTF_FORMAT_BUFFER_OVERFLOW;
+  }
+  copy_n(&fbuffer[fb_offset], fstring, fstr_len);
+  fbuffer[fb_offset + fstr_len] = '\0';
+  queue[offset].set(&fbuffer[fb_offset]);
+  return PRINTF_SUCCESS;
+}
+
 template <typename T>
-static inline void set_batch(PrintfPacket* queue, int offset, const T t) [[hc,cpu]] {
-  queue[offset].set(t);
+static inline PrintfError set_batch(PrintfPacket* queue, char* fbuffer, int offset, const T t) [[hc,cpu]] {
+  if (std::is_pointer<T>::value){
+    if (process_str_batch(queue, fbuffer, offset, (char*)t) == PRINTF_FORMAT_BUFFER_OVERFLOW)
```

Comment:
This could be an issue here.  It assumes that any pointer passing to printf is a string, which is not the case since printf supports the %p format specifier.